File: solutions/golang/digitalwallet/bank_account.go
package digitalwallet

import "math/big"

type BankAccount struct {
	BasePaymentMethod
	AccountNumber string
	RoutingNumber string
}

func NewBankAccount(id string, user *User, accountNumber, routingNumber string) *BankAccount {
	return &BankAccount{
		BasePaymentMethod: BasePaymentMethod{ID: id, User: user},
		AccountNumber:     accountNumber,
		RoutingNumber:     routingNumber,
	}
}

func (b *BankAccount) ProcessPayment(amount *big.Float, currency Currency) bool {
	// Process bank account payment
	return true
}

File: solutions/csharp/digitalwalletservice/BankAccount.cs
using System;
using System.Numerics;

namespace DigitalWallet
{
    public class BankAccount : PaymentMethod
    {
        private readonly string accountNumber;
        private readonly string routingNumber;

        public BankAccount(string id, User user, string accountNumber, string routingNumber)
            : base(id, user)
        {
            this.accountNumber = accountNumber;
            this.routingNumber = routingNumber;
        }

        public override bool ProcessPayment(decimal amount, Currency currency)
        {
            // Process bank account payment logic
            return true;
        }
    }
}